ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/RateLimitingInterceptorSpec.scala

```diff
@@ -217,6 +223,126 @@ final class RateLimitingInterceptorSpec
+  it should "limit the number of streams" in {
+
+    val metrics = new Metrics(new MetricRegistry)
+
+    val limitStreamConfig = RateLimitingConfig.Default.copy(maxStreams = 2)
+
+    val waitService = new WaitService()
+    withChannel(metrics, waitService, limitStreamConfig).use { channel: Channel =>
+      {
+        for {
+          fStatus1 <- streamHello(channel) // Ok
+          fStatus2 <- streamHello(channel) // Ok
+          activeStreams = metrics.daml.lapi.streams.active.getCount
+          fStatus3 <- streamHello(channel) // Limited
+          status3 <- fStatus3 // Closed as part of limiting
+          _ = waitService.completeStream()
+          status1 <- fStatus1
+          fStatus4 <- streamHello(channel) // Ok
+          _ = waitService.completeStream()
+          status2 <- fStatus2
+          _ = waitService.completeStream()
+          status4 <- fStatus4
+        } yield {
+          activeStreams shouldBe limitStreamConfig.maxStreams
+          status1.getCode shouldBe Code.OK
+          status2.getCode shouldBe Code.OK
+          metrics.daml.lapi.streams.active.getCount shouldBe 0
+          status3.getCode shouldBe Code.ABORTED
+          status3.getDescription should include(metrics.daml.lapi.streams.activeName)
+          status4.getCode shouldBe Code.OK
+        }
+      }
+    }
+  }
+
+  it should "exclude non-stream traffic from stream counts" in {
+
+    val metrics = new Metrics(new MetricRegistry)
+
+    val limitStreamConfig = RateLimitingConfig.Default.copy(maxStreams = 2)
+
+    val waitService = new WaitService()
+    withChannel(metrics, waitService, limitStreamConfig).use { channel: Channel =>
+      {
+        for {
+
+          fStatus1 <- streamHello(channel)
+          fHelloStatus1 = singleHello(channel)
+          fStatus2 <- streamHello(channel)
+          fHelloStatus2 = singleHello(channel)
+
+          _ = waitService.completeStream()
+          _ = waitService.completeSingle()
+          _ = waitService.completeStream()
+          _ = waitService.completeSingle()
+
+          status1 <- fStatus1
+          helloStatus1 <- fHelloStatus1
+          status2 <- fStatus2
+          helloStatus2 <- fHelloStatus2
+
+        } yield {
+          status1.getCode shouldBe Code.OK
+          helloStatus1.getCode shouldBe Code.OK
+          status2.getCode shouldBe Code.OK
+          helloStatus2.getCode shouldBe Code.OK
+          metrics.daml.lapi.streams.active.getCount shouldBe 0
+        }
+      }
+    }
+  }
+
+  it should "stream rate limiting should not limit non-stream traffic" in {
+
+    val metrics = new Metrics(new MetricRegistry)
+
+    val limitStreamConfig = RateLimitingConfig.Default.copy(maxStreams = 2)
+
+    val waitService = new WaitService()
+    withChannel(metrics, waitService, limitStreamConfig).use { channel: Channel =>
+      {
+        for {
+
+          fStatus1 <- streamHello(channel)
+          fStatus2 <- streamHello(channel)
+          fHelloStatus1 = singleHello(channel)
```

Comment:
I would add another `streamHello` right above to show that streams are limited while singles aren't

Reply:
The "limit the number of streams" test above does exactly that already.  Repeating that test here may detract from the purpose of this test.

Reply:
:+1: